Maple2.Server.Game/Manager/ShopManager.cs

```diff
@@ -0,0 +1,609 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Maple2.Database.Extensions;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Game.Shop;
+using Maple2.Model.Metadata;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Maple2.Tools.Extensions;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager;
+
+public sealed class ShopManager {
+    #region EntryId
+    private int idCounter;
+
+    /// <summary>
+    /// Generates an EntryId unique to this specific actor instance.
+    /// </summary>
+    /// <returns>Returns a local EntryId</returns>
+    private int NextEntryId() => Interlocked.Increment(ref idCounter);
+    #endregion
+    private readonly GameSession session;
+    // TODO: the CharacterShopData's restock count need to be reset at daily reset.
+    private IDictionary<int, CharacterShopData> accountShopData;
+    private IDictionary<int, CharacterShopData> characterShopData;
+    private IDictionary<int, Shop> instancedShops;
+    private IDictionary<int, IDictionary<int, CharacterShopItemData>> characterShopItemData;
+    private IDictionary<int, IDictionary<int, CharacterShopItemData>> accountShopItemData;
+    private IDictionary<int, BuyBackItem> buyBackItems;
+    private Shop? activeShop;
+    private int shopNpcId;
+
+    private readonly ILogger logger = Log.Logger.ForContext<ShopManager>();
+
+    public ShopManager(GameSession session) {
+        this.session = session;
+        instancedShops = new Dictionary<int, Shop>();
+        buyBackItems = new Dictionary<int, BuyBackItem>();
+        using GameStorage.Request db = session.GameStorage.Context();
+
+        accountShopData = new Dictionary<int, CharacterShopData>();
```

Comment:
You don't need to copy the items from one dictionary to another. If you really want a copy, you can use the copy constructor: `new Dictionary(other)`, but why can't you just use the returned value directly?